src/hotspot/share/opto/vectornode.hpp

```diff
@@ -910,6 +911,10 @@ class LoadVectorGatherNode : public LoadVectorNode {
             ((is_subword_type(vect_type()->element_basic_type())) &&
               idx == MemNode::ValueIn + 1);
   }
+  virtual int store_Opcode() const {
+    // Ensure it is different from any store opcode
+    return -1;
```

Comment:
Maybe improve this comment and explain that we are doing this to avoid folding which does not account for the mask/offsets. Same for the comments in the other `store_Opcode` methods.

Reply:
Comments updated.
Thanks @TobiHartmann!